### DISCUSSION TITLE
Spack & Petsc

### DISCUSSION THREAD

**Discussion by hbadi** [spack/spack#27355] [Q&A]
Hello,

I just discovered and installed Spack, it seems very great !

I try to install petsc with openmp+mumps depencies, but i don't understand why Spack takes a very long time installing a lot of "basic" packages (gcc9.3, automake, cmake, perl, openssl...) i already have installed in my distro (ubuntu20.04).

Moreover, now i have petsc installed, i want to install slepc with my custom petsc, how can i specify at the install stage ?

Best regards.

**Comment by lrtfm** [comment]
The tutorial is a good start: https://spack-tutorial.readthedocs.io/en/latest/tutorial_basics.html#

I think you could find how to install slepc by going through https://spack-tutorial.readthedocs.io/en/latest/tutorial_basics.html#installing-packages

```
spack install slepc ^/<hash-of-your-petsc>
```
the hash `<hash-of-your-petsc>` of your petsc can be found by
```
spack find -lf petsc
```

**Reply by hbadi** [reply]
Hello @lrtfm,

Sounds a good start ;-) It works !

**Comment by hbadi** [comment]
I'm trying to build my own package depending on PETSC, i can't see how to specify that the petsc package must be compiled with +mumps option ? I don't want to add mumps as a depend package cause my package don't know MUMPS.

**Comment by hbadi** [comment]
It seems that writing depens_on('petsc+mumps') works, but i don't know which version it uses...

**Comment by hbadi** [comment]
1 -Petsc has a variant `opemp` but it seems that nothing is installed when specifying `spack install petsc+openmp+mpi+mumps`, i get the following depencies `spack find -d petsc` :
```
petsc@3.16.1
    hdf5@1.10.7
        openmpi@4.1.1
            hwloc@2.6.0
                libpciaccess@0.16
                libxml2@2.9.12
                    libiconv@1.16
                    xz@5.2.5
                    zlib@1.2.11
                ncurses@6.2
            libevent@2.1.12
                openssl@1.1.1l
            numactl@2.0.14
            openssh@8.7p1
                libedit@3.1-20210216
        pkgconf@1.8.0
    hypre@2.23.0
        openblas@0.3.18
    metis@5.1.0
    mumps@5.4.0
        netlib-scalapack@2.1.0
        parmetis@4.0.3
    superlu-dist@7.1.1
```

[(https://petsc.org/release/miscellaneous/threads/)](https://petsc.org/release/miscellaneous/threads/)

2 - Moreover i there's no `openblas` variant for petsc ?

**Reply by lrtfm** [reply]
1. As I know, here `openmp` is an option for the compiler, not an actual package in spack.
2. Yes. PETSc depends on `blas`. You can use other packages like `atlas`, `netlib-xblas` as alternative choice:
```
spack install petsc ^atlas
```
see discuss here https://github.com/spack/spack/issues/2067

**Reply by hbadi** [reply]
Ok i got it, thanks.

**Comment by hbadi** [comment]
I try to test/install petsc with mumps and without mpi as follow :

```
user@MSI:~$ spack spec -It petsc~mpi+mumps ^mumps~mpi
Input spec
--------------------------------
 -   [    ]  petsc~mpi+mumps
 -   [    ]      ^mumps~mpi

Concretized
--------------------------------
==> Error: petsc~mpi+mumps ^mumps~mpi is unsatisfiable, conflicts are:
  A conflict was triggered
```

I also tried spack `spec -It petsc~mpi+mumps`, that's the same error, does the `~mpi` would afect `mumps` ?

Thanks.
